### ISSUE TITLE
.NET 9 RC 1 blazor webassembly standalone issue in release mode

### ISSUE THREAD

**Issue by ysmoradi** [dotnet/aspnetcore#57908]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Due the fact that there's no `@Assets[""]` in blazor webassembly standalone's index.html file, it's expected to have no files with dynamic name like the followings:
![image](https://github.com/user-attachments/assets/34155c1a-fc70-41a9-b8d0-496425a19c9d)
Note that the only file with this issue, is the isolated css bundle file of a razor class library, and the rest of the files are fine.

### Expected Behavior

The file name should be `AdminPanel.Client.Core.bundle.scp.css` instead of `AdminPanel.Client.Core.ix2lnawx70.bundle.scp`

### .NET Version

9.0.100-rc.1.24452.12

**Comment by ysmoradi** [comment]
Simply publish the attached project in local folder to see something like this:
[BlazorApp2.zip](https://github.com/user-attachments/files/17021378/BlazorApp2.zip)
![image](https://github.com/user-attachments/assets/9bcd4c70-0664-4179-ab26-c1e90d395643)
Instructions to create such a project:
1- Create blazor wasm standalone project
2- Add razor class library to it
3- Publish!

**Comment by javiercn** [comment]
@ysmoradi thanks for contacting us.

That's not how the system works. Fingerprinting can be "soft" or "hard". Files that are expected to be referenceable in user code, are "soft fingerprinted" meaning the file name doesn't change so that in the event that we are dealing with a Standalone app, things still work.

That's why for example you see `exampleJsInterop.js` is not fingerprinted so that you can reference it in code as `exampleJsInterop.js`. A similar situation happens with `<AppName>.styles.css` since that's referenced from Index.html

Files that are not directly user referenceable, like project bundles or webassembly assets are "hard" fingerprinted, which means they always get fingerprinted no matter what as all their usages are from generated code.

**Comment by ysmoradi** [comment]
Okay, and how I'm supposed to reference that hard referenced file in my index.html where therer's no `@Assets[""]` available?
And we've implemented advanced asset/cache management mechanism in our side without relying on conditional get, is there any option that we can disable that hard fingerprint in our projects?

**Comment by javiercn** [comment]
> Do you say I'm not supposed to reference that isolated css bundle in my index.html file?

Yep.

> And we've implemented advanced asset/cache management mechanism in our side without relying on conditional get, is there any option that we can disable that hard fingerprint in our projects?

Can you tell us more?

**Comment by ysmoradi** [comment]
Thanks, I just resolved the issue by correctly referencing scoped css bundle based on Microsoft Docs.

We're leveraging Blazor's PWA feature with the following enhancements:

1- Bringing PWA feature to the blazor webapp template, so it's possible to have blazor webassembly pre-rendering and PWA at the same time, being SEO friendly while improving UX with PWA features. This also runs pre-rendering only for the first user's visit, so the user will be engaged in pre-rendered contents while wasm files are being downloaded. For the next time, everything is already downloaded, so we can avoid pre-rendering by PWA taking control from the very beginging.

2- We're downloading files only once (blazor's default PWA implementation downloads/stores files two times), and we protorize downloading app's assemblies and assemblies marked as lazy + the rest of static assets will be downloaded later in background.

3- We append file's integrity hash as a query string and in both CDN and aspnetcore, we've policy of 7 days public cache without conditional get that improves performance a lot while keeping clients updated after each publish.

Live demo at https://todo.bitplatform.dev/ (Pre-render enabled PWA web app)
More info at https://bitplatform.dev/bswup/overview

P.S.
We managed to update all our websites to the .NET 9 successfully, this includes the pre-render enabled SPA website ([bitplatform.dev](bitplatform.dev)), blazor webassembly standalone web app ([adminpanel.bitplatform.dev](adminpanel.bitplatform.dev)) and blazor auto, pre-render and PWA powered web app [blazorui.bitplatform.dev](blazorui.bitplatform.dev))
